lib/watir/radio_set.rb

```diff
@@ -0,0 +1,218 @@
+module Watir
+  class RadioSet
+    extend Forwardable
+    include Watir::Exception
+    include Enumerable
+
+    delegate [:exists?, :present?, :visible?, :browser] => :source
+
+    attr_reader :source, :frame
+
+    def initialize(query_scope, selector)
+      unless selector.kind_of? Hash
+        raise ArgumentError, "invalid argument: #{selector.inspect}"
+      end
+
+      @source = Radio.new(query_scope, selector)
+      @frame = @source.parent(tag_name: :form)
+    end
+
+    #
+    # Yields each TableCell associated with this row.
+    #
+    # @example
+    #   radio_set = browser.radio_set
+    #   radio_set.each do |radio|
+    #     puts radio.text
+    #   end
+    #
+    # @yieldparam [Watir::RadioSet] element iterate through the radio buttons.
+    #
+
+    def each(&block)
+      radios.each(&block)
+    end
+
+    #
+    # Get the n'th radio button in this set
+    #
+    # @return Watir::Radio
+    #
+
+    def [](idx)
+      radios[idx]
+    end
+
+    #
+    # @return Watir::Radio
+    #
+
+    def radio(opt)
```

Comment:
I was thinking of it like a private api/method and every place this code uses it needs to pass in the opt, but you're right that isn't what this is/should be. `my_radio_set.radio` should return the first radio in that set, the same as `my_radio_set.first` or `my_radio_set[0]`